### PR TITLE
socket module

### PR DESCRIPTION
**Pull request by palaviv**
This is an initial work on an implementation of a socket module. I wanted to get some first input on my design.
You can try the code by opening a server with `nc` for example `nc -l 127.0.0.1 -p 8080` and doing the following:
```py
>>>>> import socket
>>>>> s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>>>>> s.connect("127.0.0.1:8080")
>>>>> s.send(b'aaaa')
>>>>> s.close()
```
What I wanted is to use the [`std::net`](https://doc.rust-lang.org/std/net/index.html) high level socket handlers. By saving the initial socket creation flags I can choose the correct connection object at each function.

### CODE REVIEW

File: vm/src/stdlib/socket.rs [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+                        .get_tcp_stream()
+                        .unwrap()
+                        .read_to_end(&mut buffer)
+                        .unwrap();
+                    Ok(PyObject::new(
+                        PyObjectPayload::Bytes { value: buffer },
+                        vm.ctx.bytes_type(),
+                    ))
+                }
+                _ => Err(vm.new_type_error("".to_string())),
```
Comment (windelbouwman) [inline vm/src/stdlib/socket.rs]: Instead of match you can use `if let`. Please run clippy on your code to get these kind of tips.

**Comment by windelbouwman** [comment]
I think this initial attemp is ready for merge now?

**Comment by palaviv** [comment]
I changed the code to be more nice. My next steps are:

- Add support for tcp listen
- Write some test snippets
- Add better error handling

File: tests/snippets/stdlib_socket.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+connector = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+connector.connect(("127.0.0.1", 8080))
+connection = listener.accept()[0]
+
+message_a = b'aaaa'
+message_b = b'bbbbb'
+
+connector.send(message_a)
+connector.close()
+recv_a = connection.recv(10)
```
Comment (windelbouwman) [inline tests/snippets/stdlib_socket.py]: Maybe add an assert that recv_a == message_a?

**Pull request merged**